File: src/cleansia_android/core/src/main/java/cz/cleansia/core/servicearea/ServicedCountry.kt
package cz.cleansia.core.servicearea

/**
 * Core-owned slim shape for a serviced country. Each consumer app maps
 * its NSwag-generated CountryListItem into this on its way out of the
 * adapter, so the shared `ServiceAreaProvider` doesn't carry a
 * dependency on either app's generated client package.
 *
 * [isoCode] is normalised to lowercase by the provider before exposure
 * (Mapbox's bias parameter expects `cz`, not `CZE` or `CZ`).
 */
data class ServicedCountry(
    val id: String,
    val isoCode: String,
    val name: String,
)

File: src/cleansia_ios/CleansiaCore/Sources/CleansiaCore/ServiceArea/ServicedCountry.swift
import Foundation

/// Core-owned slim shape for a serviced country — each app maps its generated
/// `CountryListItem` into this on the way out of its `ServiceAreaDataSource`,
/// so the shared provider carries no generated-client dependency (the Android
/// `core/servicearea/ServicedCountry` parity).
public struct ServicedCountry: Equatable, Sendable {
    public let id: String
    public let isoCode: String
    public let name: String

    public init(id: String, isoCode: String, name: String) {
        self.id = id
        self.isoCode = isoCode
        self.name = name
    }
}
